ansible_collections/arista/cvp/plugins/module_utils/tag_tools.py

```diff
@@ -0,0 +1,238 @@
+                        tag_val = tag['value']
+                        if mode == 'create':
+                            self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                            tag_name, tag_val)
+                        if state == 'assign':
+                            if auto_create:
+                                self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                                tag_name, tag_val)
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id)
+                        if state == 'unassign':
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id,
+                                                                       remove=True)
+                        if mode == 'delete':
+                            self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                            tag_name, tag_val,
+                                                            remove=True)
+
+        # Start build
+        request = 'REQUEST_START_BUILD'
+        reques_id = 'b1'
+        description = 'testing cvprac build'
+        self.__cv_client.api.workspace_config(workspace_id=workspace_id,
+                                              display_name=workspace_name,
+                                              description=description, request=request,
+                                              request_id=reques_id)
+
+        # Check workspace build status and proceed only after it finishes building
+        # XXX: timeout? ansible timeout!
+        b = 0
+        while b == 0:
+            build_id = reques_id
+            try:
+                request = self.__cv_client.api.workspace_build_status(workspace_id,
+                                                                      build_id)
+            except CvpRequestError:
+                continue
+            if request['value']['state'] == 'BUILD_STATE_SUCCESS':
+                b = b + 1
+            else:
+                continue
```

Comment:
I believe we can improve this section. We should also maybe have a mechanism of timeout in case the build fails.

Reply:
in this case I suggest that we look at the status message if it ever fails, for instance if we get `BUILD_STATE_FAIL` then it will not recover on its own and we should fail the play and ask the user to check what's happening on the UI. I don't expect this to ever fail for tag assignments, but we never know